### PR TITLE
Allow control of Circe formatting

### PR DESCRIPTION
**Pull request by nicolasdalsass**
Restores the ability to use a custom Printer if the user wants,
by specifying it in implicit scope.
Keeps the current default behavior of using Printer.nospaces.

### CODE REVIEW

**Comment by nicolasdalsass** [comment]
Solves #138

**Comment by hseeberger** [comment]
Nice. Yet I think I will set up benchmarks to see the impact on performance before merging. Just out of curiosity...

File: akka-http-circe/src/main/scala/de/heikoseeberger/akkahttpcirce/CirceSupport.scala [hunk 1]
```diff
@@ -140,13 +134,4 @@ trait ErrorAccumulatingUnmarshaller { this: BaseCirceSupport =>
...
-
-/**
-  * Mix-in this trait to use a compact JSON printer during marshalling.
-  */
-trait NoSpacesPrinter { this: BaseCirceSupport =>
-
-  override final def printer: Printer =
-    Printer.noSpaces
-}
+}
```
Comment (hseeberger) [inline akka-http-circe/src/main/scala/de/heikoseeberger/akkahttpcirce/CirceSupport.scala]: New line at end of file, please.

**Comment by hseeberger** [comment]
@nicolasdalsass could you please rebase your PR?

**Comment by nicolasdalsass** [comment]
Yup !

**Comment by nicolasdalsass** [comment]
Thanks for merging :-) Out of curiosity, could you spin benchmarks ? Your comment got me thinking too !

**Pull request merged**